Library/Homebrew/cmd/install-vendor.sh

```diff
@@ -0,0 +1,165 @@
+
+if [[ -n "$HOMEBREW_OSX" ]]
+then
+  if [[ "$HOMEBREW_PROCESSOR" = "Intel" ]]
+  then
+    # TODO test only. Upload files to bintray before shipping !!!
+    ruby_URL="https://github.com/Homebrew/homebrew-portable/releases/download/20160512/portable-ruby-2.0.0-p648.i386-darwin9.tar.gz"
+    ruby_SHA="035ccdc2c073172148c5ffe3e7004011862b400c6aa3a2831c1cec46a0876de5"
+    git_URL="https://github.com/Homebrew/homebrew-portable/releases/download/20160506/portable-git-2.8.2.universal-darwin9.tar.gz"
+    git_SHA="54ce39f5269202dbf08281fb7d0bbf73f03ef38b02a74e8f6ab84467bb70c793"
+  else
+    ruby_URL=""
+    ruby_SHA=""
+    git_URL=""
+    git_SHA=""
+  fi
+else
+  ruby_URL=""
+  ruby_SHA=""
+  git_URL=""
+  git_SHA=""
+fi
+
+fetch() {
+  local curl_args
+  local sha
+  local temporary_path
+
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    curl_args="-fLA"
+  elif [[ -z "$HOMEBREW_QUIET" ]]
+  then
+    curl_args="-#fLA"
+  else
+    curl_args="-sfLA"
+  fi
+
+  temporary_path="${CACHED_LOCATION}.incomplete"
+
+  mkdir -p "$HOMEBREW_CACHE"
+  [[ -z "$HOMEBREW_QUIET" ]] && echo "==> Downloading $VENDOR_URL"
+  if [[ -f "$CACHED_LOCATION" ]]
+  then
+    [[ -z "$HOMEBREW_QUIET" ]] && echo "Already downloaded: $CACHED_LOCATION"
+  else
+    if [[ -f "$temporary_path" ]]
+    then
+      "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                       -C - "$VENDOR_URL" -o "$temporary_path"
+      if [[ $? -eq 33 ]]
+      then
+        [[ -z "$HOMEBREW_QUIET" ]] && echo "Trying a full download"
+        rm -f "$temporary_path"
+        "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                         "$VENDOR_URL" -o "$temporary_path"
+      fi
+    else
+      "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                       "$VENDOR_URL" -o "$temporary_path"
+    fi
+
+    trap '' SIGINT
+    mv "$temporary_path" "$CACHED_LOCATION"
+    trap - SIGINT
+  fi
+
+  if [[ -n "$(which shasum)" ]]
+  then
+    sha="$(shasum -a 256 "$CACHED_LOCATION" | cut -d' ' -f1)"
+  elif [[ -n "$(which sha256sum)" ]]
+  then
+    sha="$(sha256sum "$CACHED_LOCATION" | cut -d' ' -f1)"
+  else
+    odie "Cannot verify the checksum ('shasum' or 'sha256sum' not found)!"
+  fi
+
+  if [[ "$sha" != "$VENDOR_SHA" ]]
+  then
+    odie <<EOS
+Checksum mismatch.
+Expected: $VENDOR_SHA
+Actual: $sha
+Archive: $CACHED_LOCATION
+To retry an incomplete download, remove the file above.
+EOS
+  fi
+}
+
+install() {
+  local tar_args
+
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    tar_args="xvzf"
+  else
+    tar_args="xzf"
+  fi
+
+  mkdir -p "$VENDOR_DIR/$VENDOR_NAME"
+  chdir "$VENDOR_DIR/$VENDOR_NAME"
+  [[ -z "$HOMEBREW_QUIET" ]] && echo "==> Install $(basename "$VENDOR_URL")"
+  trap '' SIGINT
+  tar "$tar_args" "$CACHED_LOCATION"
+  if "./$VENDOR_VERSION/bin/$VENDOR_NAME" --version >/dev/null 2>&1
+  then
+    ln -sfn "$VENDOR_VERSION" opt
+    ls -t | grep -Ev "^(opt|$VENDOR_VERSION)" | tail -n +4 | xargs rm -rf
```

Comment:
Mike askes me to keep several old copies in case of installation failure. This command comes from here. I don't know whether there is a better implementation. Will add a comment.